File: library/src/main/java/com/snail/easyble/core/BleConfig.java
package com.snail.easyble.core;

import android.bluetooth.le.ScanSettings;

/**
 * 描述:
 * 时间: 2018/8/5 18:28
 * 作者: zengfansheng
 */
public class BleConfig {
    private IScanHandler scanHandler;
    private IBondController bondController;
    private int scanPeriodMillis = 10000;
    private boolean useBluetoothLeScanner = true;
    private boolean acceptSysConnectedDevice;
    private ScanSettings scanSettings;
    private boolean hideNonBleDevice;

    /**
     * 设置扫描过滤器
     *
     * @param handler 扫描结果处理
     */
    public BleConfig setScanHandler(IScanHandler handler) {
        scanHandler = handler;
        return this;
    }

    public IScanHandler getScanHandler() {
        return scanHandler;
    }

    public int getScanPeriodMillis() {
        return scanPeriodMillis;
    }

    /**
     * 设置蓝牙扫描周期
     *
     * @param scanPeriodMillis 毫秒
     */
    public BleConfig setScanPeriodMillis(int scanPeriodMillis) {
        this.scanPeriodMillis = scanPeriodMillis;
        return this;
    }

    public boolean isUseBluetoothLeScanner() {
        return useBluetoothLeScanner;
    }

    /**
     * 控制是否使用新版的扫描器
     */
    public BleConfig setUseBluetoothLeScanner(boolean useBluetoothLeScanner) {
        this.useBluetoothLeScanner = useBluetoothLeScanner;
        return this;
    }

    public IBondController getBondController() {
        return bondController;
    }

    /**
     * 连接进行配对的控制
     */
    public BleConfig setBondController(IBondController bondController) {
        this.bondController = bondController;
        return this;
    }

    public boolean isAcceptSysConnectedDevice() {
        return acceptSysConnectedDevice;
    }

    public BleConfig setAcceptSysConnectedDevice(boolean acceptSysConnectedDevice) {
        this.acceptSysConnectedDevice = acceptSysConnectedDevice;
        return this;
    }

    public ScanSettings getScanSettings() {
        return scanSettings;
    }

    /**
     * 扫描设置
     */
    public BleConfig setScanSettings(ScanSettings scanSettings) {
        this.scanSettings = scanSettings;
        return this;
    }

    public boolean isHideNonBleDevice() {
        return hideNonBleDevice;
    }

    /**
     * 设置是否过滤非BLE设备
     */
    public void setHideNonBleDevice(boolean hideNonBleDevice) {
        this.hideNonBleDevice = hideNonBleDevice;
    }
}

File: library/src/main/java/com/snail/easyble/core/BleConfig.kt
package com.snail.easyble.core

import com.snail.easyble.annotation.RunOn

/**
 *
 * date: 2018/8/5 18:28
 * author: zengfansheng
 */
open class BleConfig {
    /**
     * 搜索设置
     */
    open var scanConfig = ScanConfig()
    /**
     * 是否在连接时配对
     */
    open var bondController: IBondController? = null
    /**
     * [Device]实例构建器，搜索到BLE设备时，使用此构建器实例化[Device]
     */
    open var deviceCreator: IDeviceCreator? = null
    /**
     * 数据发布者（被观察者）
     */
    open var eventObservable: EventObservable = EventObservable()
    /**
     * 观察者或者回调的方法在没有使用注解指定调用线程时，默认被调用的线程
     */
    open var methodDefaultInvokeThread = RunOn.POSTING
    /**
     * 自动重连时，搜索次数与间隔的对应关系，key：已尝试次数，value：间隔，单位为毫秒。如搜索了1次，间隔2秒，搜索了5次，间隔30秒等
     */
    open var scanIntervalPairsInAutoReonnection = mutableListOf(Pair(0, 2000), Pair(1, 5000), Pair(3, 10000), Pair(5, 30000), Pair(10, 60000))
}
